src/ui/Icon/Icon.ts

```diff
@@ -93,6 +102,16 @@ export class Icon extends Component {
     this.result = this.result || this.resolveResult();
     Assert.exists(this.result);
 
+    if (TemplateFieldsEvaluator.evaluateFieldsToMatch(this.options.conditions, this.result)) {
+      this.initialize(element, bindings);
+    } else {
+      if (this.element.parentElement != null) {
+        this.element.parentElement.removeChild(this.element);
```

Comment:
Same question as further above: when a condition doesn't match, do we want to remove the option or have a fallback?